src/app/components/audios/audios.component.ts

```diff
@@ -1,9 +1,114 @@
   standalone: true,
+  imports: [
+    DataTableComponent,
+    LoadingComponent,
+    TabsComponent,
+    CommonModule,
+    TableRowTemplateDirective,
+    RouterLink,
+    TabDirective,
+  ],
 })
-export class AudiosComponent {}
+export class AudiosComponent {
+  audioService = inject(AudioService);
+  datePipe = inject(DatePipe);
+
+  userId: number;
+
+  audioRequests$: Observable<UserAudioRequest[]>;
+  expiredAudioRequests$: Observable<UserAudioRequest[]>;
+
+  inProgessRows$: Observable<UserAudioRequestRow[]>;
+  completedRows$: Observable<UserAudioRequestRow[]>;
+  expiredRows$: Observable<UserAudioRequestRow[]>;
+
+  data$: Observable<{
+    inProgessRows: UserAudioRequestRow[];
+    completedRows: UserAudioRequestRow[];
+    expiredRows: UserAudioRequestRow[];
+  }>;
+
+  columns = [
+    { name: 'Case ID', prop: 'caseNumber', sortable: true },
+    { name: 'Court', prop: 'courthouse', sortable: true },
+    { name: 'Hearing date', prop: 'hearingDate', sortable: true },
+    { name: 'Start time', prop: 'startTime', sortable: true },
+    { name: 'End time', prop: 'endTime', sortable: true },
+    { name: 'Request ID', prop: 'requestId', sortable: true },
+    { name: 'Expiry date', prop: 'expiry', sortable: true },
+    { name: 'Status', prop: 'status', sortable: true },
+  ];
+
+  constructor(private route: ActivatedRoute) {
+    this.userId = this.route.snapshot.data.userState.userId;
+
+    this.audioRequests$ = this.audioService.getAudioRequestsForUser(this.userId, false);
+    this.expiredAudioRequests$ = this.audioService.getAudioRequestsForUser(this.userId, true);
+
+    this.inProgessRows$ = this.audioRequests$.pipe(
+      map((audioRequests) => this.filterInProgressRequests(audioRequests)),
+      map((audioRequests) => this.mapAudioRequestToRows(audioRequests))
+    );
+
+    this.completedRows$ = this.audioRequests$.pipe(
+      map((audioRequests) => this.filterCompletedRequests(audioRequests)),
+      map((audioRequests) => this.mapAudioRequestToRows(audioRequests))
+    );
+
+    this.expiredRows$ = this.expiredAudioRequests$.pipe(
+      map((audioRequests) => this.mapAudioRequestToRows(audioRequests))
+    );
+
+    this.data$ = combineLatest({
+      inProgessRows: this.inProgessRows$,
+      completedRows: this.completedRows$,
+      expiredRows: this.expiredRows$,
+    });
+  }
+
+  mapAudioRequestToRows(audioRequests: UserAudioRequest[]): UserAudioRequestRow[] {
+    return audioRequests.map((ar) => {
+      return {
+        // id: ar.case_id,
+        caseNumber: ar.case_number,
+        courthouse: ar.courthouse_name,
+        hearingDate: this.datePipe.transform(ar.hearing_date, 'dd MMM yyyy'),
+        startTime: this.datePipe.transform(ar.media_request_start_ts, 'hh:mm:ss', 'utc'),
+        endTime: this.datePipe.transform(ar.media_request_end_ts, 'hh:mm:ss', 'utc'),
+        requestId: ar.media_request_id,
+        expiry: this.datePipe.transform(ar.media_request_expiry_ts, 'hh:mm:ss dd/mm/yyyy', 'utc'),
```

Comment:
Is it better to use the pipe here or in the template?

Reply:
I did it this way initially as the table didn't support the dynamic templating / ability to add datepipe in the table template. Doing it here in the ts mapping allows me to do it once or potentially 3 times in the template. Having said that, I have a template partial that I reuse now so I don't think if it matters either way apart from preference or readability I guess.